e3nn/spherical_tensor.py

```diff
@@ -90,3 +93,232 @@ def spherical_harmonics_coeff_to_sphere(coeff, alpha, beta):
     ls = list(range(lmax + 1))
     sh = rsh.spherical_harmonics_alpha_beta(ls, alpha, beta)
     return torch.einsum('...i,i->...', sh, coeff)
+
+
+def projection(vectors, lmax, sum_points=True, radius=True):
+    radii = vectors.norm(2, -1)
+    vectors = vectors[radii > 0.]
+
+    if radius:
+        radii = radii[radii > 0.]
+    else:
+        radii = torch.ones_like(radii[radii > 0.])
+
+    angles = o3.xyz_to_angles(vectors)
+    coeff = spherical_harmonics_dirac(lmax, *angles)
+    coeff *= radii.unsqueeze(-1)
+    return coeff.sum(-2) if sum_points else coeff
+
+
+def adjusted_projection(vectors, lmax, sum_points=True, radius=True):
+    radii = vectors.norm(2, -1)
+    vectors = vectors[radii > 0.]
+    angles = o3.xyz_to_angles(vectors)
+
+    coeff = projection(vectors, lmax, sum_points=False, radius=radius)
+    A = torch.einsum("ai,bi->ab", (rsh.spherical_harmonics_alpha_beta(list(range(lmax + 1)), *angles), coeff))
+    try:
+        coeff *= torch.lstsq(radii, A).solution.view(-1).unsqueeze(-1)
+    except:
+        coeff *= torch.gels(radii, A).solution.view(-1).unsqueeze(-1)
+    return coeff.sum(-2) if sum_points else coeff
+
+
+class SphericalTensor():
+    def __init__(self, signal, mul, lmax):
+        self.signal = signal
+        self.lmax = lmax
+        self.mul = mul
+        self.Rs = rs.convention([(mul, l) for l in range(lmax + 1)])
```

Comment:
One should allow to specify the parity.
A signal on the sphere can have 4 different parities (or not defined).
`f(r) ---> p0 * f(u * r)` where p0 and u are +-1.
where parity of `signal_L` is `p=p0 * u^L`

Reply:
I don't understand. You might need to make the edit yourself as I'm not sure how this would effect plotting.